src/Microsoft.ML.Transforms/Text/TextCatalog.cs

```diff
@@ -98,6 +98,9 @@ public static class TextCatalog
 
         /// <summary>
         /// Extracts word embeddings.
+        /// Output three times more values than dimension of the model specified in <paramref name="modelKind"/>
+        /// First set of values represent minumum encountered values (for each dimension), second set represent average (for each dimension)
```

Comment:
Without checking the code, I'm unsure of the ordering of min/avg/max. If you haven't I'd check the code to ensure this is right.

Reply:
for defaults:
```
editor.Values[i] = float.MaxValue;
editor.Values[i + dimension] = 0;
editor.Values[i + offset] = float.MinValue;
```
during iteration over words:
```
float currentTerm = wordVector[i];
if (editor.Values[i] > currentTerm)
    editor.Values[i] = currentTerm;
editor.Values[dimension + i] += currentTerm;
if (editor.Values[offset + i] < currentTerm)
    editor.Values[offset + i] = currentTerm;

if (deno != 0)
    for (int index = 0; index < dimension; index++)
        editor.Values[index + dimension] /= deno;

```


it's Min/Average/Max

---
In reply to: [252511622](https://github.com/dotnet/machinelearning/pull/2339#discussion_r252511622) [](ancestors = 252511622)